### ISSUE TITLE
Fix S2589 FP: Var pattern should not raise

### ISSUE THREAD

**Issue by Tim-Pohlmann** [SonarSource/sonar-dotnet#8326]
```
void Test(int i, object o)
{
    _ = i switch
    {
        1 => 1,
        var other => 2  // Noncompliant FP: var should not raise
    };
            
    _ = o switch
    {
        1 => 1,
        var other => 2  // Compliant: by accident, o has no SymbolicValue and always branches du to #8324
    };
}
```
The `var` pattern always matches, which is why it is raising S2589. While this is technically correct (the best kind of correct), it does not make sense to raise here, since the point of the pattern is to be a catch-all.
When implementing an exception, make sure that the rule about unreachable code (S2583) still raises, if necessary:
```
void Test(int i)
{
    if (i is var x1)            // Noncompliant: unreachable code
        Console.WriteLine();
    else
        Console.WriteLine();    // Secondary
}
```
